spring-cloud-gcp-pubsub-stream-binder/src/test/java/com/google/cloud/spring/stream/binder/pubsub/provisioning/PubSubChannelProvisionerTests.java

```diff
@@ -289,4 +236,56 @@ void testProvisionConsumerDestination_recursiveExistCalls() {
     assertThatExceptionOfType(ProvisioningException.class)
         .isThrownBy(() -> this.pubSubChannelProvisioner.ensureTopicExists("new_topic", true));
   }
+
+  @Test
+  void testProvisionConsumerDestination_subscriptionNameCannotBeNull() {
+    when(this.pubSubConsumerProperties.isAutoCreateResources()).thenReturn(false);
+    when(this.pubSubConsumerProperties.getSubscriptionName()).thenReturn(null);
+    assertThatExceptionOfType(IllegalArgumentException.class)
+        .isThrownBy(
+            () ->
+                this.pubSubChannelProvisioner.provisionConsumerDestination(
+                    "topic_A", null, this.properties))
+        .withMessage("Subscription Name cannot be null or empty");
+  }
+
+  @Test
+  void testProvisionConsumerDestination_createAnonymousSubscription() {
```

Comment:
How's this different from `testProvisionConsumerDestination_anonymousGroup()` above?

Reply:
correct, `testProvisionConsumerDestination_createAnonymousSubscription` is not required